document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/handler/CompanyReportDataHandler.java

```diff
@@ -0,0 +1,134 @@
+
+    public DocumentInfoResponse getCompanyReport(String resourceUri, String requestId)
+        throws HandlerException {
+
+        String companyNumber = getCompanyNumberFromUri(resourceUri);
+
+        try {
+            LOG.infoContext(requestId, "Getting data for report for company number: " + companyNumber, getDebugMap(companyNumber));
+            return createDocumentInfoResponse(companyNumber);
+        } catch (URIValidationException | ApiErrorResponseException | MapperException e) {
+            LOG.errorContext(requestId,"Failed to get data for report for company number " + companyNumber, e, getDebugMap(companyNumber));
+            throw new HandlerException(e.getMessage(), e.getCause());
+        }
+    }
+
+    private DocumentInfoResponse createDocumentInfoResponse(String companyNumber) throws HandlerException,
+        URIValidationException, ApiErrorResponseException, MapperException {
+
+        DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+
+        documentInfoResponse.setData(getCompanyReportData(companyNumber));
+        documentInfoResponse.setAssetId("company-report");
+        documentInfoResponse.setPath(createPathString());
+        documentInfoResponse.setTemplateName("company-report.html");
+
+        return documentInfoResponse;
+    }
+
+    private String getCompanyReportData(String companyNumber) throws HandlerException,
+        URIValidationException, ApiErrorResponseException, MapperException {
+
+        CompanyReportApiData companyReportApiData = new CompanyReportApiData();
+
+        CompanyProfileApi companyProfileApi = getCompanyProfile(companyNumber);
+
+        companyReportApiData.setCompanyProfileApi(companyProfileApi);
+
+        return toJson(companyReportMapper.mapCompanyReport(companyReportApiData));
+    }
+
+
+    private String toJson(CompanyReport companyReport) throws HandlerException {
+
+        String reportToJson;
+        ObjectMapper mapper = new ObjectMapper();
+        JavaTimeModule javaTimeModule = new JavaTimeModule();
+        javaTimeModule.addDeserializer(LocalDate.class, new LocalDateDeserializer(DateTimeFormatter.ofPattern("dd-MMMM-yyyy")));
+        javaTimeModule.addSerializer(LocalDate.class, new LocalDateSerializer(DateTimeFormatter.ofPattern("dd-MMMM-yyyy")));
+        mapper.registerModule(javaTimeModule);
+        mapper.setDateFormat(new SimpleDateFormat("dd-MMMM-yyyy"));
+
+        try {
+            LOG.info("Attempting to convert company report to JSON");
+            reportToJson = mapper.writeValueAsString(companyReport);
+        } catch (JsonProcessingException e) {
+            throw new HandlerException(
+                "Could not serialise Document Info for the company report for");
+        }
+
+        return reportToJson;
+    }
+
+    private CompanyProfileApi getCompanyProfile(String companyNumber) throws HandlerException {
+
+        try {
+            LOG.info("Attempting to retrieve company profile");
```

Comment:
swap to infoContext and add requestId and a debug map so Logs can be tracked for the specific reque